src/printer.js

```diff
@@ -987,9 +987,24 @@ function genericPrintNoParens(path, options, print) {
       return group(concat(parts));
     case "ForStatement": {
       const body = adjustClause(path.call(print, "body"), options);
+      const hasSomeDanglingLineComment = n.comments &&
+        n.comments.some(comment => {
+          return !comment.leading && !comment.trailing &&
+            comment.type === "Line"
+        });
 
       if (!n.init && !n.test && !n.update) {
-        return concat(["for (;;)", body]);
+        return concat([
+          "for ",
+          hasSomeDanglingLineComment
+            ? concat([
+              comments.printDanglingComments(path, options, true),
+              line
+            ])
+            : comments.printDanglingComments(path, options, true),
```

Comment:
What happens if you just put this condition in? The comment is not going to appear at the same place but if it is stable, then this is okay.